packages/core/core/src/migrations/database/5.0.0-discard-drafts.ts

```diff
@@ -113,13 +113,22 @@ export async function* getBatchToDiscard({
   db,
   trx,
   uid,
-  batchSize = 500,
+  defaultBatchSize = 1000,
 }: {
   db: Database;
   trx: Knex;
   uid: string;
-  batchSize?: number;
+  defaultBatchSize?: number;
 }) {
+  const client = db.config.connection.client;
+  const isSQLite =
+    typeof client === 'string' && ['sqlite', 'sqlite3', 'better-sqlite3'].includes(client);
+
+  // The SQLite documentation states that the maximum number of terms in a
+  // compound SELECT statement is 500 by default.
+  // See: https://www.sqlite.org/limits.html
+  // To ensure a successful migration, we limit the batch size to 500 for SQLite.
+  const batchSize = isSQLite ? 500 : defaultBatchSize;
```

Comment:
Shouldn't this be the following to avoid overriding a lower batch size on SQLite (manually defined)? 

```suggestion
  const batchSize = isSQLite ? Math.min(defaultBatchSize, 500) : defaultBatchSize;
```